tests/spec/fs.unlink.spec.js

```diff
@@ -104,3 +104,41 @@ describe('fs.unlink', function() {
     });
   });
 });
+
+/**
+ * fsPromises tests
+ */
+
+describe('fsPromises.unlink', function() {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should not unlink directories', function (done) {
+    var fs = util.fs().promises;
+
+    fs.mkdir('/mydir')
+      .then(
+        error => {
+          if(error) throw error;
+        });
+
+    fs.unlink('/mydir')
```

Comment:
See point 1 - Chain function calls with `.then` to ensure proper execution order.